google/data_source_google_compute_instance_group_test.go

```diff
@@ -193,14 +193,19 @@ func testAccCheckDataSourceGoogleComputeInstanceGroup(dataSourceName string) res
 
 func testAccCheckDataSourceGoogleComputeInstanceGroupConfig() string {
 	return fmt.Sprintf(`
+data "google_compute_image" "my_image" {
+       family  = "debian-9"
+       project = "debian-cloud"
+}
+
 resource "google_compute_instance" "test" {
   name         = "tf-test-%s"
   machine_type = "n1-standard-1"
   zone         = "us-central1-a"
 
   boot_disk {
     initialize_params {
-      image = "debian-cloud/debian-8"
+      image = "${data.google_compute_image.my_image.self_link}"
```

Comment:
I'm fine with using the data source, but I'm not totally sure why it's a better idea than just saying `image = "debian-cloud/debian-9"` (which feels more realistic with what our users are likely to be doing). Can you explain further?

Reply:
I was trying to use the data source because then the disk image can't change without it being recorded in Terraform. For instance templates and groups especially, this seems like the intended usage. I was trying to standardise, perhaps foolishly. For instances and disks I could see switching back, and don't mind doing that, if that's what you'd prefer.